debezium-core/src/main/java/io/debezium/processors/reselect/ReselectColumnsPostProcessor.java

```diff
@@ -250,10 +250,8 @@ private boolean isUnavailableValueHolder(Schema schema, Object value) {
             case MAP:
                 return unavailableValuePlaceholderMap.equals(value);
             case STRING:
-                if (Json.LOGICAL_NAME.equals(schema.name())) {
-                    return unavailableValuePlaceholderJson.equals(value);
-                }
-                return unavailableValuePlaceholder.equals(value);
+                final boolean isJsonAndUnavailable = Json.LOGICAL_NAME.equals(schema.name()) && unavailableValuePlaceholderJson.equals(value);
+                return unavailableValuePlaceholder.equals(value) || isJsonAndUnavailable;
```

Comment:
As discussed on Zulip, could you add a few comment lines here describing the why for this unique check, specifically calling out how we serialize HSTORE vs JSON/JSONB for PostgreSQL, just so we know precisely why this needs to be this way without having to re-evaluate the source code again.  Thanks.

Reply:
I added some description there, let me know if it is clear enough